app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
-                          .fillMaxSize()) {
-                    Spacer(modifier = Modifier.height(6.dp))
-                    // Display of the buttons
-                    Row {
-                      // Display of the Ingredients button
-                      Button(
-                          onClick = { ingredientsView = true },
-                          shape = RoundedCornerShape(0.dp),
-                          modifier = Modifier.width(150.dp).testTag("ingredientsButton"),
-                          colors =
-                              ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
-                          border = BorderStroke(2.dp, Color.Black)) {
-                            Text(
-                                text = "Ingredients",
-                                fontSize = 12.sp,
-                                textAlign = TextAlign.Center)
-                          }
-                      // Display of the Instruction buttons
-                      Button(
-                          onClick = { ingredientsView = false },
-                          shape = RoundedCornerShape(0.dp),
-                          modifier = Modifier.width(150.dp).testTag("instructionsButton"),
-                          colors =
-                              ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
-                          border = BorderStroke(2.dp, Color.Black)) {
-                            Text(
-                                text = "Instructions",
-                                fontSize = 12.sp,
-                                textAlign = TextAlign.Center)
-                          }
-                    }
-                    // Display of the list of ingredients with the ability to change the number of
-                    // servings
-                    IngredientInstructionView(ingredientsView, servingsCount, currentRecipe) {
-                        newServingsCount ->
-                      servingsCount = newServingsCount
-                    }
-                  }
+                    .padding(PADDING.dp)
+                    .verticalScroll(rememberScrollState())) {
+              if (currentRecipe == null) {
+                Text(
+                    text = LOADING,
+                    style = MaterialTheme.typography.bodyMedium,
+                    color = MaterialTheme.colorScheme.onSecondary)
+              } else {
+                RecipeImage(currentRecipe!!)
+                RecipeDescription(currentRecipe!!)
+                PrepareCookTotalTimeDisplay()
+                RecipeInformation(currentRecipe!!)
+              }
+            }
+      })
+}
+
+/**
+ * Display of the recipe information
+ *
+ * @param currentRecipe: The recipe to display
+ */
+@Composable
+private fun RecipeInformation(currentRecipe: Recipe) {
+  Column(
+      horizontalAlignment = Alignment.CenterHorizontally,
```

Comment:
### Nice Work
- **Toggle Setup**: Love the `isIngredientDisplay` toggle setup—keeps the UI flexible so users can easily switch between ingredients and instructions. Nice touch!

### A Few Ideas

1. **`fillMaxSize()` on the `Column`**:
   - Right now, `fillMaxSize()` on the main `Column` might stretch the background more than you need. If the goal is just to fit around the content, maybe try `wrapContentHeight()` instead to keep things a bit tighter.

   **Quick Fix**:
   ```kotlin
   modifier = Modifier
       .background(MaterialTheme.colorScheme.primary, shape = RoundedCornerShape(OVERVIEW_RECIPE_ROUND))
       .wrapContentHeight()
       .padding(SMALL_PADDING.dp)
   ```

2. **Toggle Logic Readability**:
   - The toggle logic works well, but it might read even better if the `isIngredientDisplay` flag directly matches the button label’s purpose. Could simplify it by making the `SlidingButton` calls a little more descriptive.

   **Idea**:
   ```kotlin
   SlidingButton(
           isIngredientDisplay = isIngredientDisplay,
           text = stringResource(R.string.ingredients),
           testTag = "ingredientButton",
           modifier = Modifier.weight(1f),
           clickable = { isIngredientDisplay = true }
       )

       SlidingButton(
           isIngredientDisplay = !isIngredientDisplay, // Reads clearer with label’s actual state
           text = stringResource(R.string.instructions),
           testTag = "instructionsButton",
           modifier = Modifier.weight(1f),
           clickable = { isIngredientDisplay = false }
       )
   ```

3. **Accessibility on Toggle Buttons**:
   - Might be worth adding a `contentDescription` (put it in `strings.xml`) on `SlidingButton` for screen readers, so they know which button is which.

   **Example**:
   ```kotlin
   Text(
       text = text,
       style = MaterialTheme.typography.bodyMedium,
       color = if (!isIngredientDisplay) MaterialTheme.colorScheme.background else MaterialTheme.colorScheme.onPrimary,
       textAlign = TextAlign.Center,
       modifier = Modifier
           .padding(SMALL_PADDING.dp)
           .testTag(testTag)
           .semantics { contentDescription = text } // Adds accessibility
   )
   ```

Let me know if you want to dig into more details or look at other sections!